packages/astro/src/vite-plugin-astro-server/index.ts

```diff
@@ -215,13 +215,14 @@ async function handleRequest(
 
 	let body: ArrayBuffer | undefined = undefined;
 	if (!(req.method === 'GET' || req.method === 'HEAD')) {
-		let bytes: string[] = [];
+		let bytes: Uint8Array[] = [];
 		await new Promise((resolve) => {
-			req.setEncoding('utf-8');
-			req.on('data', (bts) => bytes.push(bts));
+			req.on('data', part => {
+				bytes.push(part);
+			});
 			req.on('end', resolve);
 		});
-		body = new TextEncoder().encode(bytes.join('')).buffer;
+		body = Buffer.concat(bytes);
```

Comment:
Are we OK to use `Buffer` here? I know that's a Node-ism, but I guess `astro dev` only ever runs in Node-compatible runtimes?

Reply:
Yeah, all of the vite plugins only run in Node